src/main/java/bio/terra/service/dataset/flight/ingest/DatasetIngestFlight.java

```diff
@@ -98,6 +99,15 @@ public DatasetIngestFlight(FlightMap inputParameters, Object applicationContext)
     RetryRule lockDatasetRetry =
         getDefaultRandomBackoffRetryRule(appConfig.getMaxStairwayThreads());
 
+    // Add these steps to clear out scratch file if the flight has failed
+    if (cloudPlatform.isGcp()) {
+      addStep(new PerformPayloadIngestStep(new IngestLandingFileDeleteGcpStep(true, gcsPdao)));
+    } else if (cloudPlatform.isAzure()) {
+      addStep(
+          new PerformPayloadIngestStep(
+              new IngestLandingFileDeleteAzureStep(true, azureContainerPdao)));
+    }
+
```

Comment:
I thought we had decided not to do this, since there's already a TTL on the bucket, and always deleting the file would make debugging very difficult.

Reply:
Sadness.